src/gc/gc.d

```diff
@@ -605,61 +605,72 @@ struct GC
             else
             {
                 auto pool = gcx.findPool(p);
-                psize = pool.getSize(p);     // get allocated size
-                if (psize >= PAGESIZE && size >= PAGESIZE)
+                if (pool.isLargeObject)
```

Comment:
As an aside, I'm looking at this multipage `if` and thinking, better kill me now...